arcaflow_plugin_stressng/stressng_schema.py

```diff
@@ -728,6 +728,17 @@ class StressNGParams:
         ),
     ] = None
 
+    taskset: typing.Annotated[
+        typing.Optional[str],
+        validation.pattern(re.compile(r"((\d{1,3})(\,|\-))*\d{1,3}")),
```

Comment:
This regex, while clever in certain ways, has a couple of drawbacks.

First, I believe that having the wildcard on the leading tokens of the expression means that the matcher could potentially undergo significant backtracking.  Admittedly, the damage is limited in this case, because the wildcard subexpression (and the overall expression) is fairly straightforward.  However, as a general rule, you want to construct your expression to avoid backtracking to the best extent possible.  (For instance, according to [https://regex101.com/](https://regex101.com/), matching `1` with your regex takes eleven steps and matching `1-2,3` takes twenty-six (after I added anchoring); whereas, my suggestion below requires eight and twenty, respectively.)

Second, your regex will match `1-2-3`, which I don't think you want it to.

Beyond those, your regex is lacking anchoring, which I think will make it match other things that you don't want it to.  And, its using _capturing groups_ which it doesn't need to (because they aren't referenced).

So, I recommend this regex instead:
```
^(?:\d{1,3}|\d{1,3}-\d{1,3})(?:,(?:\d{1,3}|\d{1,3}-\d{1,3}))*$
```
That is, a group of either one to three digits or one to three digits followed by a hyphen followed by one to three digits, then zero or more groups consisting of a comma followed by a group like the previous.

You can consider coding this as follows:
```python
    taskgroup = r"(?:\d{1,3}|\d{1,3}-\d{1,3})"
    taskset: typing.Annotated[
        typing.Optional[str],
        validation.pattern(re.compile(f"^{taskgroup}(?:,{taskgroup})*$")
```

Reply:
It's interesting to me that your regex can be shortened to `^((\d{1,3})|(?2)-(?2))(?:,(?1))*$`, but this results in more steps.

Reply:
Trying your expression on https://regex101.com/, the expression itself doesn't parse if I select Python regexes.  (And, Python doesn't like it either.)

Still, I can't explain why it takes more steps.  However, I would be concerned that `(?2)` might be trying to match the _value_ of the first (er, second) group rather than trying to reuse its pattern.  But, I think that's all moot.

Reply:
> I can't explain why it takes more steps.

Of course, that site has a "debugging" which will tell you what the steps _are_...but I'm working on other stuff....

Reply:
Overlooked this change while working on the tests and other problems I came across. Fixing it now...

Reply:
Done.